var/spack/repos/builtin/packages/aocl-sparse/package.py

```diff
@@ -0,0 +1,115 @@
+# Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+import os
+import inspect
+
+
+class AoclSparse(CMakePackage):
+    """AOCL-Sparse is a library that contains basic linear algebra subroutines
+    for sparse matrices and vectors optimized for AMD family of processors.
+    It is designed to be used with C and C++. Current functionality of sparse
+    library supports SPMV function with CSR and ELLPACK formats."""
+
+    homepage = "https://developer.amd.com/amd-aocl/aocl-sparse/"
+    url = "https://github.com/amd/aocl-sparse/archive/3.0.tar.gz"
+    git = "https://github.com/amd/aocl-sparse.git"
+
+    maintainers = ['amd-toolchain-support']
+
+    version('3.0',  sha256='9c280a4be164b6ea257b4575ec3c55f154d4dc5a8d2d929a73eb683173bcc95c')
+    version('2.2',  sha256='33c2ed6622cda61d2613ee63ff12c116a6cd209c62e54307b8fde986cd65f664')
+
+    conflicts("%gcc@:9.1.999", msg="Minimum required GCC version is 9.2.0")
+
+    variant('build_type', default='Release',
+            description='CMake build type',
+            values=('Debug', 'Release'))
+    variant('shared', default=True,
+            description='Build shared library')
+    variant('ilp64', default=False,
+            description='Build with ILP64 support')
+
+    depends_on('boost', when='@2.2')
+
+    @property
+    def build_directory(self):
+        """Returns the directory to use when building the package
+
+        :return: directory where to build the package
+        """
+
+        builddir = self.stage.source_path
+
+        if self.spec.variants['build_type'].value == 'Debug':
+            builddir = join_path(self.stage.source_path, 'build', 'debug')
+        else:
+            builddir = join_path(self.stage.source_path, 'build', 'release')
+
+        mkdirp(builddir)
+        return builddir
+
+    def cmake(self, spec, prefix):
```

Comment:
Why not implement `cmake_args` to create and return the list of arguments and leave lines 86-87 to the base class?